tensorflow/contrib/seastar/seastar_tensor_coding.cc

```diff
@@ -0,0 +1,38 @@
+
+#include <pthread.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include "tensorflow/contrib/seastar/seastar_tensor_coding.h"
+#include "tensorflow/core/common_runtime/device.h"
+
+namespace tensorflow {
```

Comment:
I feel like all the (de-)serialization code could be moved here instead of scattering around in multiple locations.

Reply:
yes, i'll do some refactoring work, including your suggestions.